backend/server/controllers/posts.js

```diff
@@ -0,0 +1,150 @@
+import models from '../../database/models'
+import { Authentication } from '../../middlewares/authentication'
+
+/*
+Requires authentication.
+Response codes:
+201 CREATED when the post has successfully been created.
+500 INTERNAL SERVER ERROR otherwise.
+*/
+export const createPost = async (req, res) => {
+    try {
+        const authToken = req.get('Authorization')
+        const { body } = req
+
+        if (!authToken) {
+            res.status(400).send({
+                'Error message': 'Auth token not provided',
+            })
+        } else {
+            const decodedUser = Authentication.extractUser(authToken)
+
+            if (!decodedUser.id) {
+                res.status(401).send({ 'Error message': 'Auth token invalid' })
+            }
+
+            /*
+            NOTE: attachments is currently a STRING type as the attachment functionality has not been setup yet.
+            */
+            const createNewPost = await models.posts.create({
+                text_content: body.text_content,
+                author: decodedUser.username,
+                parent: body.parent,
+                usersLiked: 0,
+                usersShared: 0,
+                attachments: '',
+            })
+            res.status(201).send(createNewPost)
+        }
+    } catch (error) {
+        res.status(500).send(error)
+    }
+}
+
+/*
+Does not require authentication.
+Path paramter: id - the id of the post that is being retrieved.
+Response Codes:
+200 OK when the post has been successfully found.
+404 NOT FOUND when the post with that id can not be found.
+500 INTERNAL SERVER ERROR for everything else.
+*/
+export const getPostById = async (req, res) => {
+    try {
+        const { params } = req
```

Comment:
Should check the request is authorized first (all endpoints require authorization except the create users endpoint)

Reply:
I think it will be better to not have authentication for viewing posts

Reply:
Michael said "not for now probs", so I'm just going to resolve this conversation.

Reply:
yeah allg